src/udev/udevadm-info.c

```diff
@@ -254,6 +254,53 @@ static void cleanup_dir(DIR *dir, mode_t mask, int depth) {
         }
 }
 
+/*
+ * Assume that dir is a directory with file names matching udev data base
+ * entries for devices in /run/udev/data (such as "b8:16"), and removes
+ * all files except those that haven't been deleted in /run/udev/data
+ * (i.e. they were skipped during db cleanup because of the db_persist flag).
+ * Returns true if the directory is empty after cleanup.
+ */
+static bool cleanup_dir_after_db_cleanup(DIR *dir, DIR *datadir) {
+        unsigned int kept = 0;
+
```

Comment:
assert on dir and datadir

Reply:
> assert on dir and datadir

It's a static function and the caller, which checks NULL, is just a few lines below. Is this *really* necessary?

Reply:
It may not necessary, but it improves readability of the code.

Reply:
Yes - coding style, it's in-line documentation for callers. You know this know, but in 5 years after three refactors when it's called from six different places it won't be so obvious anymore

Reply:
resolved with b148f0cc26